{"ast":null,"code":"import _asyncToGenerator from \"/Users/ybezeha/WebstormProjects/new-project/node_modules/@angular-devkit/build-angular/node_modules/@babel/runtime/helpers/esm/asyncToGenerator.js\";\n// pkg/dist-src/auth.js\nfunction auth(_x) {\n  return _auth.apply(this, arguments);\n} // pkg/dist-src/is-rate-limit-error.js\nfunction _auth() {\n  _auth = _asyncToGenerator(function* (reason) {\n    return {\n      type: \"unauthenticated\",\n      reason\n    };\n  });\n  return _auth.apply(this, arguments);\n}\nimport { RequestError } from \"@octokit/request-error\";\nfunction isRateLimitError(error) {\n  if (error.status !== 403) {\n    return false;\n  }\n  if (!error.response) {\n    return false;\n  }\n  return error.response.headers[\"x-ratelimit-remaining\"] === \"0\";\n}\n\n// pkg/dist-src/is-abuse-limit-error.js\nimport { RequestError as RequestError2 } from \"@octokit/request-error\";\nvar REGEX_ABUSE_LIMIT_MESSAGE = /\\babuse\\b/i;\nfunction isAbuseLimitError(error) {\n  if (error.status !== 403) {\n    return false;\n  }\n  return REGEX_ABUSE_LIMIT_MESSAGE.test(error.message);\n}\n\n// pkg/dist-src/hook.js\nfunction hook(_x2, _x3, _x4, _x5) {\n  return _hook.apply(this, arguments);\n} // pkg/dist-src/index.js\nfunction _hook() {\n  _hook = _asyncToGenerator(function* (reason, request, route, parameters) {\n    const endpoint = request.endpoint.merge(route, parameters);\n    return request(endpoint).catch(error => {\n      if (error.status === 404) {\n        error.message = `Not found. May be due to lack of authentication. Reason: ${reason}`;\n        throw error;\n      }\n      if (isRateLimitError(error)) {\n        error.message = `API rate limit exceeded. This maybe caused by the lack of authentication. Reason: ${reason}`;\n        throw error;\n      }\n      if (isAbuseLimitError(error)) {\n        error.message = `You have triggered an abuse detection mechanism. This maybe caused by the lack of authentication. Reason: ${reason}`;\n        throw error;\n      }\n      if (error.status === 401) {\n        error.message = `Unauthorized. \"${endpoint.method} ${endpoint.url}\" failed most likely due to lack of authentication. Reason: ${reason}`;\n        throw error;\n      }\n      if (error.status >= 400 && error.status < 500) {\n        error.message = error.message.replace(/\\.?$/, `. May be caused by lack of authentication (${reason}).`);\n      }\n      throw error;\n    });\n  });\n  return _hook.apply(this, arguments);\n}\nvar createUnauthenticatedAuth = function createUnauthenticatedAuth2(options) {\n  if (!options || !options.reason) {\n    throw new Error(\"[@octokit/auth-unauthenticated] No reason passed to createUnauthenticatedAuth\");\n  }\n  return Object.assign(auth.bind(null, options.reason), {\n    hook: hook.bind(null, options.reason)\n  });\n};\nexport { createUnauthenticatedAuth };","map":null,"metadata":{},"sourceType":"module","externalDependencies":[]}